xdsl/dialects/experimental/stencil.py

```diff
@@ -200,13 +111,90 @@ def max(a: IndexAttr, b: IndexAttr | None) -> IndexAttr:
     def as_tuple(self) -> tuple[int, ...]:
         return tuple(e.data for e in self.array.data)
 
+    def as_list(self) -> list[int]:
```

Comment:
what for?

Reply:
QoL!
We have duck-typing instead of a proper ShapeType going on, which returns a shape as a list[int], it was helping.
I can see if I can get rid of the tuple[int] if the both of them bother you?

Reply:
`tuple[int]` is the better one, though, I think we should make the duck typing accept tuples instead

Reply:
This should not live here but rather in some tools/utils file.
It is totally unrelated to the stencil dialect, but also something that can be used in other places as well

Reply:
I disagree with @georgebisbas here, I think we want this on the class. Also, we can move this to `tuple[int, ...]`, but that's for another PR because we'd also need to change the memref type then.

Reply:
Opened #1008 for this